graphx/src/main/scala/org/apache/spark/graphx/impl/VertexPartitionBaseOps.scala

```diff
@@ -136,6 +136,31 @@ private[graphx] abstract class VertexPartitionBaseOps
     leftJoin(createUsingIndex(other))(f)
   }
 
+  def leftJoinWithFold[VD2: ClassTag, VD3: ClassTag, A]
+      (other: Self[VD2], acc: A)
```

Comment:
Reduce-style aggregation is named 'leftJoin', and fold-style aggregation is 'leftJoinWithFold'.
A rule of naming is kind of inconsistent, so would it be better to rename 'leftJoin' into 'leftJoinWithReduce'? Thought?